tests/backends/hardware/sr/v4/test_power_board.py

```diff
@@ -0,0 +1,470 @@
+        wIndex: int = 0,
+        data_or_wLength: Optional[Union[int, bytes]] = None,
+        timeout: Optional[int] = None,
+    ) -> bytes:
+        """Mock reading data from a device."""
+        assert wValue == 0  # Always 0 on read.
+
+        if 0 <= wIndex < 6:
+            return self.read_output(data_or_wLength)
+        if wIndex == 7:
+            return self.read_battery(data_or_wLength)
+        if wIndex == 8:
+            return self.read_button(data_or_wLength)
+        if wIndex == 9:
+            return self.read_fw(data_or_wLength)
+
+        raise NotImplementedError
+
+    def read_output(self, data_or_wLength: Union[int, bytes]):
+        """Mock reading the output current."""
+        assert int(data_or_wLength) == 4
+        return struct.pack("<I", 1200)
+
+    def read_battery(self, data_or_wLength: Union[int, bytes]):
+        """Mock reading the battery sensor."""
+        assert int(data_or_wLength) == 8
+        return struct.pack("<II", 567, 982)
+
+    def read_button(self, data_or_wLength: Union[int, bytes]):
+        """Mock reading the button state."""
+        assert int(data_or_wLength) == 4
+        return struct.pack("<I", 0)  # Not Pressed
+
+    def read_fw(self, data_or_wLength: Union[int, bytes]):
+        """Mock reading the firmware number."""
+        assert int(data_or_wLength) == 4
+        return struct.pack("<I", self.firmware_version)
+
+    def write_data(
+        self,
+        wValue: int = 0,
+        wIndex: int = 0,
+        data_or_wLength: Optional[Union[int, bytes]] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        """Mock writing data to a device."""
+        if 0 <= wIndex < 6:
+            # Write Output.
+            return self.write_bool(wValue, data_or_wLength)
+        if wIndex == 6:
+            return self.write_bool(wValue, data_or_wLength)
+        if wIndex == 7:
+            return self.write_bool(wValue, data_or_wLength)
+        if wIndex == 8:
+            # Buzz the Piezo
+            return self.write_buzz(wValue, data_or_wLength)
+
+        raise NotImplementedError
+
+    def write_bool(self, wValue: int, data_or_wLength: Union[int, bytes]):
+        """Pretend to write a bool."""
+        assert wValue == 1 or wValue == 0
+        assert data_or_wLength == b""
+
+    def write_buzz(self, wValue: int, data_or_wLength: Union[int, bytes]):
+        """Pretend to buzz."""
+        assert wValue == 0
+        frequency, duration_ms = struct.unpack("<HH", data_or_wLength)
+        assert frequency >= 0
+        assert duration_ms >= 0
+
+
+def mock_find(
+    find_all=True, *, idVendor: int, idProduct: int,
+) -> List[MockUSBPowerBoardDevice]:
+    """This function mocks the behaviour of usb.core.find."""
+    assert idVendor == 0x1BDA
+    assert idProduct == 0x0010
+    if find_all:
+        return [MockUSBPowerBoardDevice(f"SERIAL{n}") for n in range(0, 4)]
+    else:
+        return [MockUSBPowerBoardDevice("SERIAL0")]
```

Comment:
I don't forsee us calling `find(find_all=False)` in the near future, so this `else` block is redundant. I'd suggest just asserting `find_all == True` instead.

Code that never gets executed is doing nothing but waiting to get out of date.